premenv.nix

```diff
@@ -15,18 +15,20 @@ stdenv.override (stdenvPrev: {
           if [[ -z "$goldLicense" ]]; then
             printf "Don't want ads? Buy a \033[33;1mnixpkgs gold\033[0m license for a premium experience.\n"
             sleep 3
-            NIXPKGS_GOLD_AD=$(shuf -n 1 ${./ads.txt})
+
+            # Export the ad to advertise it to the build environment
+            export NIXPKGS_GOLD_AD="$(echo -n "$NIXPKGS_GOLD_ADS" | shuf -n1)"
             NIXPKGS_GOLD_AD_LENGTH=''$((''${#NIXPKGS_GOLD_AD}))
             printf "\033[2;5m|  |"
             printf " %.0s" $(seq 1 $NIXPKGS_GOLD_AD_LENGTH)
             printf "|  |\033[0m\n"
             printf "\033[2;5m|AD|\033[0m"
             echo "$NIXPKGS_GOLD_AD" | tr -d "\n" | sed 's/nixpkgs gold/\x1b[33;1mnixpkgs gold\x1b[0m/g'
             printf "\033[2;5m|AD|\033[0m\n"
-            sleep 10
             printf "\033[2;5m|  |"
             printf " %.0s" $(seq 1 $NIXPKGS_GOLD_AD_LENGTH)
             printf "|  |\033[0m\n"
+            sleep 10
```

Comment:
The placement of this `sleep` is intentional so that it follows the ad. Otherwise, the ad isn't visible without `-L`.